sky/backends/provision_utils.py

```diff
@@ -263,25 +263,17 @@ def _wait_ssh_connection_indirect(
         ssh_private_key: str,
         ssh_control_name: Optional[str] = None,
         ssh_proxy_command: Optional[str] = None) -> bool:
-    del ssh_control_name
     # We test ssh with 'echo', because it is of the most common
     # commandline programs on both Unix-like and Windows platforms.
     # NOTE: Ray uses 'uptime' command and 10s timeout.
-    command = [
-        'ssh',
-        '-T',
-        '-i',
+
+    command = command_runner.ssh_options_list(
         ssh_private_key,
-        f'{ssh_user}@{ip}',
-        '-o',
-        'StrictHostKeyChecking=no',
-        '-o',
-        'ConnectTimeout=20s',
-    ]
-    if ssh_proxy_command is not None:
-        command += ['-o', f'ProxyCommand={ssh_proxy_command}']
-    command += ['echo']
-    proc = subprocess.run(command,
+        ssh_control_name=ssh_control_name,
```

Comment:
We probably don’t want the control path as it may cause some issue with VMs failed to connect.

Also, it would be nice to get rid of ForwardAgent.

Do you want to add the -T back